core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -308,27 +308,29 @@ class Partition(val topicPartition: TopicPartition,
                 s"different from the requested log dir $logDir")
             false
           case None =>
-            createLogIfNotExists(isNew = false, isFutureReplica = true, highWatermarkCheckpoints)
+            // not sure if topic ID should be none here, but not sure if we have access in ReplicaManager where this is called.
+            // could also use topicId method here potentially. This is only used in ReplicaManager (ZK code) so probably ok to set as None.
+            createLogIfNotExists(isNew = false, isFutureReplica = true, highWatermarkCheckpoints, None)
             true
         }
       }
     }
   }
 
-  def createLogIfNotExists(isNew: Boolean, isFutureReplica: Boolean, offsetCheckpoints: OffsetCheckpoints): Unit = {
+  def createLogIfNotExists(isNew: Boolean, isFutureReplica: Boolean, offsetCheckpoints: OffsetCheckpoints, topicId: Option[Uuid] = None): Unit = {
```

Comment:
We usually try to avoid optional arguments. The problem is that they can be easily overlooked.

Reply:
I was a little worried about this too. Some methods (not this one) are called in many locations (mostly tests) so I didn't want to bloat the changelog. But I can change these all to be explicit.